### DISCUSSION TITLE
How would you write any operator in Supabase rest client?

### DISCUSSION THREAD

**Discussion by Nithur-M** [supabase/supabase#27579] [Questions]
For example, this query: select * from someTable where '%someInput%' LIKE ANY(someColum);

**Comment by GaryAustin1** [comment]
https://supabase.com/blog/postgrest-11-1-release#anyall-filter-modifiers

**Reply by Nithur-M** [reply]
This is awesome. 

How would you use this in this structure: 
```
.or(`company.ilike.%${query}%, title.ilike.%${query}%, tags.ilikeanyof.[%${query}%]`)`
```

**Reply by GaryAustin1** [reply]
You have to use raw PostgREST format in the .or().
https://postgrest.org/en/v12/references/api/tables_views.html#operator-modifiers
It shows that PostgREST uses (any) as a modify on the filtername.  So probably `tags.ilike(any).{%value%,%value2}'  `
Note that arrays are {} not [] in Postgres.  

Edit:  It also shows available of * in place of % because of URL encoding.  Not sure if that is needed from the Supabase client or not.

**Reply by Nithur-M** [reply]
This is not working. FWIW, the tags column is a text array, any pointers?

**Reply by GaryAustin1** [reply]
If you hard code some values does it work?  As noted you may need to use * in place of %.
Also your initial code seems like it would surround the array with %,%.  I would think you need that around each item in the array.

**Reply by GaryAustin1** [reply]
OH.   If tags is array then that will not work.
ilike only works on a single value column.

**Reply by GaryAustin1** [reply]
You would need to use array operators:
![image](https://github.com/supabase/supabase/assets/54564956/928a3eee-0a0d-48e7-830f-30669f6dc8a3)
But there is no ability to do partials with those.

**Reply by Nithur-M** [reply]
Oh interesting, thanks.

Do you have any other suggestion to tackle this?

**Reply by GaryAustin1** [reply]
If your tags array column is reasonable size you could do a generated column that converts it into a comma separated string.  And do the ilike on that...

**Reply by Nithur-M** [reply]
can you provide a sample code for that? thanks

**Reply by GaryAustin1** [reply]
This is how to do a generated column.  It will add space to the database (hence my comment about reasonable size).
https://www.postgresql.org/docs/current/ddl-generated-columns.html

You can look up sql to convert an array column to a string as well as I can or ask an AI.  I don't code enough regularly to have sql in for such in my head.
